tajo-storage/tajo-storage-hdfs/src/main/java/org/apache/tajo/storage/FileTablespace.java

```diff
@@ -315,15 +315,21 @@ public boolean accept(Path path) {
     }
   }
 
+  protected List<FileStatus> listStatus(Path... dirs) throws IOException {
+    return listStatus(false, dirs);
+  }
+
   /**
    * List input directories.
    * Subclasses may override to, e.g., select only files matching a regular
    * expression.
    *
+   * @param check if target table is a partitioned table
+   * @param input directories
    * @return array of FileStatus objects
    * @throws IOException if zero items.
    */
-  protected List<FileStatus> listStatus(Path... dirs) throws IOException {
+  protected List<FileStatus> listStatus(boolean hasPartition, Path... dirs) throws IOException {
```

Comment:
This method is declared as ```protected``` and not used only in ```getSplits()``` method. So, I think that we don't have to maintain two separate methods according to the ```hasPartition``` flag.